src/lib/rest/rest.cpp

```diff
@@ -906,32 +906,33 @@ bool urlCheck(ConnectionInfo* ciP, const std::string& url)
 *
 * This function returns the version of the API for the incoming message,
 * based on the URL.
-* If the URL starts with "/v2" then the request is considered API version 2.
 *
-* Otherwise, API version 1.
+* /v2    -> 2
+* /v1    -> 1
+* /admin -> 0
 *
-* Except ...
-* The new request to change the log level (not trace level), uses the
-* URL /admin/log, which DOES NOT start with '/v2', but as some render methods
-* depend on the apiVersion and we prefer the 'new render' from v2 for this
-* operation (see OrionError::render), we consider internally /admin/log to be part of v2.
+* Otherwise, -1 is returned, corresponding to invalid path cases.
 *
-* FIXME P2: instead of looking at apiVersion for rendering, perhaps we need
-*           some other algorithm, considering 'admin' requests as well ...
 */
-static std::string apiVersionGet(const char* path)
+static int apiVersionGet(const char* path)
 {
   if ((path[1] == 'v') && (path[2] == '2'))
   {
-    return "v2";
+    return 2;
+  }
+
+  // Different from v2, v1 is case-insensitive (see case/2057 test)
+  if (((path[1] == 'v') || (path[1] == 'V')) && (path[2] == '1'))
+  {
+    return 1;
   }
 
-  if (strcmp(path, "/admin/log") == 0)
+  if (strncmp("/admin", path, strlen("/admin")) == 0)
   {
-    return "admin";
+    return 0;
   }
 
-  return "v1";
+  return -1;
 }
 
```

Comment:
I think the criteria should be:

```
/v2 -> 2
/v1 -> 1
other admin ops -> 0
uknown -> -1

Looking to `apiVersionGet()` operation

```
  if (strncmp("/admin", path, strlen("/admin")) == 0)
  {
    return 0;
  }
```

it happens that `/version` and `/statistics` are outside `/admin` so they get -1. However, it seems it doesn't have any effect (as .test regresion works 100%) so I understand that the version number is not used in that cases.

Worth the time to research on it?